Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/TvMainActivity.java

```diff
@@ -195,6 +201,41 @@ public void launchOpenFileActivity(int requestCode)
     startActivityForResult(intent, requestCode);
   }
 
+  @Override
+  public void launchOnlineUpdate()
+  {
+    if (WiiUtils.isSystemMenuInstalled())
+    {
+      SystemUpdateViewModel viewModel = new ViewModelProvider(this).get(SystemUpdateViewModel.class);
+      viewModel.setRegion(0);
+      OnlineUpdateProgressBarDialogFragment progressBarFragment = new OnlineUpdateProgressBarDialogFragment();
+      progressBarFragment.show(getSupportFragmentManager(), "OnlineUpdateProgressBarDialogFragment");
+      progressBarFragment.setCancelable(false);
+    }
+    else
+      {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        OnlineUpdateRegionSelectDialogFragment dialogFragment = new OnlineUpdateRegionSelectDialogFragment();
+        dialogFragment.show(fragmentManager, "OnlineUpdateRegionSelectDialogFragment");
+      }
+  }
```

Comment:
`launchOnlineUpdate` and `launchWiiSystemMenu` look very similar across MainActivity and TvMainActivity. Could they be put in MainPresenter instead?